### PR TITLE
create ui, Fetch and display data

### PR DESCRIPTION
**Pull request by nang2002hv**
## Related Tickets
- [#Task 78572: Content](https://edu-redmine.sun-asterisk.vn/issues/78572)
## WHAT
Create UI for Search screen, Fetch and display data

## Evidence (Screenshot or Video)
anh1ha

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/blob/master/eng/android/coding_convention_android_kotlin.md | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(Other notes)*

### CODE REVIEW

**Review by shinnosuke14202 — APPROVED** [review]

**Review by nhikiu — APPROVED** [review]

File: app/src/main/java/com/example/book/screen/search/adapter/SearchAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            binding.tvTitle.text = book.title
+            Glide.with(binding.root.context)
+                .load(book.image)
+                .into(binding.imgBook)
+        }
+    }
+
+    fun updateData(listBook: List<Book>) {
+        this.listBook = listBook
+        println("listBook: $listBook")
```
Comment (ghost) [inline app/src/main/java/com/example/book/screen/search/adapter/SearchAdapter.kt]: Không log ra kiểu này em ơi. Em dùng Log.d để có thể tự động bỏ log khi build bản release nhé

```suggestion
```

File: app/src/main/java/com/example/book/screen/search/adapter/SearchAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    override fun getItemCount(): Int {
+        return listBook.size
+    }
+
+    class ViewHolder(
+        private val binding: ItemSearchBookBinding,
+    ) : RecyclerView.ViewHolder(binding.root) {
+        fun bind(book: Book) {
+            println("book1: $book")
```
Comment (ghost) [inline app/src/main/java/com/example/book/screen/search/adapter/SearchAdapter.kt]:
```suggestion
```

**Review by ghost — CHANGES_REQUESTED** [review]
check my cmt

File: app/src/main/java/com/example/book/screen/search/SearchFragment.kt [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    override fun onDestroy() {
+        super.onDestroy()
+        presenter.onStop()
+    }
+
+    fun initSearchView() {
+        val searchView = binding.searchView
+        val searchText = searchView.findViewById<TextView>(androidx.appcompat.R.id.search_src_text)
+        searchText.setHintTextColor(ContextCompat.getColor(requireContext(), R.color.color_search_hint))
+        searchText.setTextSize(TypedValue.COMPLEX_UNIT_SP, 14f)
```
Comment (HVgiang86) [inline app/src/main/java/com/example/book/screen/search/SearchFragment.kt]: Không hard code thế này. Em khai báo trong dimen và dùng getResource

**Review by HVgiang86 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**